raspberry/SmartHome/arduinodevices/src/main/java/smarthome/raspberry/arduinodevices/server/httphandlers/BaseRequestHandler.kt

```diff
@@ -18,10 +14,9 @@ internal abstract class BaseRequestHandler : RequestHandler {
         return NanoHTTPD.Response(NanoHTTPD.Response.Status.BAD_REQUEST, MIME_PLAINTEXT, "Invalid request: $message")
     }
 
-    protected fun getController(params: Map<String, String>): BaseController {
-        // todo add checks so it won't crash
+    protected suspend fun getController(params: Map<String, String>): BaseController {
         val controllerGuid = params.getValue("controller_guid").toLong()
-        return input.findController(controllerGuid)
+        return output.findController(controllerGuid)
```

Comment:
DeviceChannelOutput does not have findController function

Reply:
it does